jib-core/src/test/java/com/google/cloud/tools/jib/cache/DefaultCacheStorageWriterTest.java

```diff
@@ -99,6 +108,85 @@ public void testWrite_uncompressed() throws IOException {
     Assert.assertEquals(layerDigest.getHash(), Blobs.writeToString(Blobs.from(selectorFile)));
   }
 
+  @Test
+  public void testWriteMetadata_v21()
+      throws IOException, URISyntaxException, InvalidImageReferenceException {
+    Path manifestJsonFile =
+        Paths.get(getClass().getClassLoader().getResource("core/json/v21manifest.json").toURI());
+    V21ManifestTemplate manifestTemplate =
+        JsonTemplateMapper.readJsonFromFile(manifestJsonFile, V21ManifestTemplate.class);
+    ImageReference imageReference = ImageReference.parse("image.reference/project/thing:tag");
+
+    new DefaultCacheStorageWriter(defaultCacheStorageFiles)
+        .writeMetadata(imageReference, manifestTemplate);
+
+    Assert.assertTrue(
+        Files.exists(
+            defaultCacheStorageFiles
+                .getImageDirectory(imageReference)
+                .resolve("manifest.amd64.linux.json")));
+
+    V21ManifestTemplate expectedManifest =
+        JsonTemplateMapper.readJsonFromFile(
+            defaultCacheStorageFiles
+                .getImageDirectory(imageReference)
+                .resolve("manifest.amd64.linux.json"),
+            V21ManifestTemplate.class);
+    Assert.assertEquals(
+        expectedManifest.getContainerConfiguration().get().getArchitecture(),
+        manifestTemplate.getContainerConfiguration().get().getArchitecture());
+  }
+
+  @Test
+  public void testWriteMetadata_v22()
+      throws IOException, URISyntaxException, InvalidImageReferenceException {
+    Path containerConfigurationJsonFile =
+        Paths.get(
+            getClass().getClassLoader().getResource("core/json/containerconfig.json").toURI());
+    ContainerConfigurationTemplate containerConfigurationTemplate =
+        JsonTemplateMapper.readJsonFromFile(
+            containerConfigurationJsonFile, ContainerConfigurationTemplate.class);
+    Path manifestJsonFile =
+        Paths.get(getClass().getClassLoader().getResource("core/json/v22manifest.json").toURI());
+    BuildableManifestTemplate manifestTemplate =
+        JsonTemplateMapper.readJsonFromFile(manifestJsonFile, V22ManifestTemplate.class);
+    ImageReference imageReference = ImageReference.parse("image.reference/project/thing:tag");
+
+    new DefaultCacheStorageWriter(defaultCacheStorageFiles)
+        .writeMetadata(imageReference, manifestTemplate, containerConfigurationTemplate);
+
+    Assert.assertTrue(
+        Files.exists(
+            defaultCacheStorageFiles
+                .getImageDirectory(imageReference)
```

Comment:
For a similar reason, I often prefer using a constant for expected values like `Files.exists(Paths.get(cacheRoot, "sub", "directory", "manifest.wasm.js.json"))`. For example, suppose `getImageDirectory()` became buggy and always returns just the cache root. The test will pass because `writeMedata()` will use the same buggy `getImageDirectory()`.